### PR TITLE
fix and test for overlapping stream subscription on update

### PR DESCRIPTION
**Pull request by tbeets**
In a non-clustered JetStream system, subject overlap with other streams in the account was not checked on stream update.

/cc @nats-io/core

### CODE REVIEW

File: server/stream.go [hunk 1]
```diff
@@ -1434,6 +1452,11 @@ func (mset *stream) update(config *StreamConfig) error {
 
 // Update will allow certain configuration properties of an existing stream to be updated.
 func (mset *stream) updateWithAdvisory(config *StreamConfig, sendAdvisory bool) error {
+	_, jsa, err := mset.acc.checkForJetStream()
```
Comment (derekcollison) [inline server/stream.go:1455]: Can we assume mset.acc non nil here? Can't remember off top of my head.

File: server/stream.go [hunk 2]
```diff
@@ -1444,6 +1467,13 @@ func (mset *stream) updateWithAdvisory(config *StreamConfig, sendAdvisory bool)
 		return NewJSStreamInvalidConfigError(err, Unless(err))
 	}
 
+	jsa.mu.Lock()
```
Comment (derekcollison) [inline server/stream.go:1470]: This an RW? Can we use RLock?

File: server/stream.go (hunk 1 above)
Comment (tbeets) [reply server/stream.go:1455]: There is an mset.account() accessor function.  Better to use that and check for nil return?

Comment (derekcollison) [reply server/stream.go:1455]: I think so.

File: server/stream.go (hunk 2 above)
Comment (tbeets) [reply server/stream.go:1470]: Yes, we can use RLock here I will change.   subjectsOverlap() is called under Lock but that is a side-effect of single caller's RW critical section.

**Comment by tbeets** [comment]
@derekcollison fixing update validation will reject existing streams on server start that already overlap.  Update is called on restore if any assignments or config have possibly changed (not deep equal).

```
[80468] 2022/12/15 10:56:42.550434 [INF]   Starting restore for stream '$G > 8576ea68-70b6-4155-a8fc-a21e86c18878'
[80468] 2022/12/15 10:56:42.550521 [WRN]   Error recreating stream "8576ea68-70b6-4155-a8fc-a21e86c18878": subjects overlap with an existing stream (10065)
```

**Comment by tbeets** [comment]
Note that the issue with offending streams (overlapping sub with other stream) failing to recreate on server startup is a current **v2.9.9** issue for non-JS clustered environments only.  This PR plugs a validation hole that can create that scenario in the first place.

**Review by derekcollison — APPROVED** [review]
LGTM

**Pull request merged**